src/BenchmarksServer/Startup.cs

```diff
@@ -266,5 +281,35 @@ private static Process StartProcess(string hostname, string benchmarksRepo, Serv
 
             return process;
         }
+
+        private static string ComputeServerUrl(string hostname, Scenario scenario)
+        {
+            var scenarioName = scenario.ToString();
+            var path = scenarioName;
+
+            var field = scenario.GetType().GetTypeInfo().GetField(scenarioName);
+            var pathAttribute = field.GetCustomAttribute<ScenarioPathAttribute>();
+            if (pathAttribute != null)
+            {
+                path = pathAttribute.Paths.First().Trim('/');
+            }
+
+            return $"http://{hostname}:5000/{path.ToLower()}";
+        }
+
+        private class SourceRepoComparer : IEqualityComparer<Source>
+        {
+            public static readonly SourceRepoComparer Instance = new SourceRepoComparer();
+
+            public bool Equals(Source x, Source y)
+            {
+                return string.Equals(x.Repository, y.Repository);
```

Comment:
As mentioned above, I would use a weaker equality check that just compares the repo name (after last slash and before last dot).